### DISCUSSION TITLE
Could not find a declaration file for module

### DISCUSSION THREAD

**Discussion by blessanm86** [TypeStrong/ts-node#1906] [General]
### Search Terms
`Could not find a declaration file for module 'json-to-pretty-yaml'`

I am trying to upgrade from v9.1.1 to v10.9.1
I keep getting the above error but my I have a `.d.ts` file in the directory for this module. It worked totally fine with the older version.

I tried many things like adding types, typeRoots and paths in the tsconfig. Nothing seems to have any affect.
The code works fine if I run tsc. 

This is my tsconfig
```
{
  "extends": "../../../tsconfig.base.json",
  "compilerOptions": {
    "module": "commonjs",
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "noImplicitOverride": true,
    "noPropertyAccessFromIndexSignature": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    }
  ]
}
```

### Specifications

```
ts-node v10.9.1
node v16.10.0
compiler v4.8.4
```

* Operating system and version: Mac Monterey

**Comment by cspotcode** [comment]
The easiest fix is to enable the `files` option.

https://typestrong.org/ts-node/docs/options#files

Add to tsconfig:

```json
"ts-node": {"files": true},
```

**Comment by blessanm86** [comment]
Sorry forgot to mention. I actually did try that option. Still didn't work.
I reference a `tsconfig.lib.json` in the `tsconfig.json`

```
{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../../dist/out-tsc",
    "declaration": true,
    "types": []
  },
  "include": ["**/*.ts"],
  "exclude": ["jest.config.ts", "**/*.spec.ts", "**/*.test.ts"]
}
```

This should include all `.ts` files in the directory. Is there anything else I should do for it to find a `.d.ts` file?

**Comment by blessanm86** [comment]
Ended up adding this `/// <reference types="./json-to-pretty-yaml" />` to the file where its used
